integrations/go/workload_executor_test.go

```diff
@@ -0,0 +1,153 @@
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/integration/mtest"
+	"go.mongodb.org/mongo-driver/mongo/integration/unified"
+	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
+)
+
+// hasLoop returns whether or not testCase tc contains a loop operation
+func hasLoop(tc *unified.TestCase) bool {
+	if tc == nil {
+		return false
+	}
+	for _, op := range tc.Operations {
+		if op.Name == "loop" {
+			return true
+		}
+	}
+	return false
+}
+
+// marshalStructToFile marshals the given object and writes to filePath
+func marshalStructToFile(t *testing.T, obj interface{}, filePath string) {
+	// MarshalExtJSON is used to print out bson.Raw properly
+	marshaled, err := bson.MarshalExtJSON(obj, false, false)
+	if err != nil {
+		t.Fatalf("marshal results failed: %v", err)
+	}
+	err = ioutil.WriteFile(filePath, marshaled, 0644)
+	if err != nil {
+		t.Fatalf("write to file failed: %v", err)
+	}
+}
+
+func TestAtlasPlannedMaintenance(t *testing.T) {
+	connstring := os.Args[1]
+	workloadSpec := []byte(os.Args[2])
+
+	setupOpts := mtest.NewSetupOptions().SetURI(connstring)
+	if err := mtest.Setup(setupOpts); err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if err := mtest.Teardown(); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	// killAllSessions will return an auth error if it's run
+	fileReqs, testCases := unified.ParseTestFile(t, workloadSpec, unified.NewOptions().SetRunKillAllSessions(false))
+
+	mtOpts := mtest.NewOptions().
+		RunOn(fileReqs...).
+		CreateClient(false)
+	mt := mtest.New(t, mtOpts)
+	defer mt.Close()
+
+	// a testfile can parse to multiple testCases, though we only expect to get one per file
+	for _, testCase := range testCases {
```

Comment:
I mentioned this in my previous set of comments, but the comment is now outdated: how is the executor expected to handle multiple test cases in a single file? In your response, you linked to the part of the unified test format spec that says "Tests SHOULD NOT include multiple loop operations (nested or sequential)", but that only covers a single test case. What if the test file parses to two separate test cases, each of which has one `loop` operation?